### PR TITLE
feat(998): append to history on install --replace

### PR DESCRIPTION
**Pull request by fibonacci1729**
closes issue #1456 

Append to release history on install replace.

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/helm/1456)

### CODE REVIEW

File: cmd/tiller/release_server.go [hunk 1]
```diff
@@ -603,6 +607,7 @@ func (s *releaseServer) prepareRelease(req *services.InstallReleaseRequest) (*re
 
 	name, err := s.uniqName(req.Name, req.ReuseName)
 	if err != nil {
+		log.Printf("FAILING HERE: %s\n", err)
```
Comment (fibonacci1729) [inline cmd/tiller/release_server.go]: (nit): remove debug log stmt

File: cmd/tiller/release_server.go [hunk 2]
```diff
@@ -737,22 +742,53 @@ func (s *releaseServer) performRelease(r *release.Release, req *services.Install
...
+				log.Printf("warning: Release replace %q failed: %s", r.Name, err)
+				old.Info.Status.Code = release.Status_SUPERSEDED
+				r.Info.Status.Code = release.Status_FAILED
+				s.recordRelease(old, true)
+				s.recordRelease(r, false)
+				return res, err
+			}
+
+		default:
+			// nothing to replace
```
Comment (fibonacci1729) [inline cmd/tiller/release_server.go]: In the default case, (i.e. no previous release, but 'replace' flag was provided), the resource should still be created as in the else block.

**Comment by technosophos** [comment]
manually testing

**Comment by technosophos** [comment]
There is still a race condition when Kubernetes does not finish deleting a resource before the next `helm install --name...` comes in. But that's sort of outside the scope of this fix. This fix does work as intended.

**Pull request merged**